app/lib/actions/katello/upstream_subscriptions/delete.rb

```diff
@@ -0,0 +1,30 @@
+module Actions
+  module Katello
+    module UpstreamSubscriptions
+      class Delete < Actions::Base
+        def plan(organization, entitlement_ids)
+          sequence do
+            entitlement_ids.each do |id|
+              plan_self(organization_id: organization.id, entitlement_id: id)
```

Comment:
does this actually work?  Can you plan_self multiple times?    I've not tried it.  Normally we would plan some other action like this, rather than plan_self in a loop.